internal/timewarp/timewarp_test.go

```diff
@@ -0,0 +1,266 @@
+								},
+								"versions": {
+									"1.0.0": {
+										"version": "1.0.0",
+										"description": "v1 desc",
+										"repository": "repo1"
+									},
+									"2.0.0": {
+										"version": "2.0.0",
+										"description": "v2 desc",
+										"repository": "repo2"
+									}
+								}
+							}`)),
+						},
+					},
+				},
+			},
+			want: &http.Response{
+				StatusCode: http.StatusOK,
+				Header: http.Header{
+					"Content-Type": []string{"application/json"},
+				},
+				Body: io.NopCloser(bytes.NewBufferString(`{
+					"time": {
+						"created": "2021-01-01T00:00:00Z",
+						"modified": "2021-06-01T00:00:00Z",
+						"1.0.0": "2021-06-01T00:00:00Z"
+					},
+					"versions": {
+						"1.0.0": {
+							"version": "1.0.0",
+							"description": "v1 desc",
+							"repository": "repo1"
+						}
+					},
+					"description": "v1 desc",
+					"repository": "repo1",
+					"dist-tags": {
+						"latest": "1.0.0"
+					}
+				}`)),
+			},
+		},
+		{
+			name:      "pypi project request - successful time warp",
+			url:       "http://localhost:8081/some-package",
+			basicAuth: "pypi:2022-01-01T00:00:00Z",
+			client: &httpxtest.MockClient{
+				Calls: []httpxtest.Call{
+					{
+						Method: "GET",
+						URL:    "https://pypi.org/simple/some-package",
+						Response: &http.Response{
+							StatusCode: http.StatusOK,
+							Header: http.Header{
+								"Content-Type": []string{"application/json"},
+							},
+							Body: io.NopCloser(bytes.NewBufferString(`{
+								"info": {
+									"name": "some-package"
+								},
+								"releases": {
+									"1.0.0": [
+										{
+											"upload_time_iso_8601": "2021-06-01T00:00:00Z",
+											"filename": "some-package-1.0.0.tar.gz"
+										}
+									],
+									"2.0.0": [
+										{
+											"upload_time_iso_8601": "2022-06-01T00:00:00Z",
+											"filename": "some-package-2.0.0.tar.gz"
+										}
+									]
+								}
+							}`)),
+						},
+					},
+					{
+						Method: "GET",
+						URL:    "https://pypi.org/simple/pypi/some-package/1.0.0/json",
+						Response: &http.Response{
+							StatusCode: http.StatusOK,
+							Header: http.Header{
+								"Content-Type": []string{"application/json"},
+							},
+							Body: io.NopCloser(bytes.NewBufferString(`{
+								"info": {
+									"name": "some-package",
+									"version": "1.0.0",
+									"requires_dist": ["req1", "req2"]
+								}
+							}`)),
+						},
+					},
+				},
+			},
+			want: &http.Response{
+				StatusCode: http.StatusOK,
+				Header: http.Header{
+					"Content-Type": []string{"application/json"},
+				},
+				Body: io.NopCloser(bytes.NewBufferString(`{
+					"info": {
+						"name": "some-package",
+						"version": "1.0.0",
+						"requires_dist": ["req1", "req2"]
+					},
+					"releases": {
+						"1.0.0": [
+							{
+								"upload_time_iso_8601": "2021-06-01T00:00:00Z",
+								"filename": "some-package-1.0.0.tar.gz"
+							}
+						]
+					}
```

Comment:
Is the response a combination of the release `info` object and the package `releases` list?